toptal/picasso
# [FX-4594] Migrate Grid to TailwindCSS

[FX-4594]

### Description

This pull request migrates Picasso grid to TailwindCSS.

Logic replicates the https://github.com/mui/material-ui/blob/v4.x/packages/material-ui/src/Grid/Grid.js.

### How to test


- [Temploy](https://picasso.toptal.net/fx-4594-migrate-griditem)
- Check out Staff Portal pull request https://github.com/toptal/staff-portal/pull/12661
  - Manual tests
    - compare https://t-staff-portal-pr-12661.toptal.net/dashboard and https://staff-portal.toptal.net/dashboard
    - compare https://t-staff-portal-pr-12661.toptal.net/talents and https://staff-portal.toptal.net/talents 
    - compare https://t-staff-portal-pr-12661.toptal.net/talents/4849690 and https://staff-portal.toptal.net/talents/4849690
    - compare https://t-staff-portal-pr-12661.toptal.net/clients and https://staff-portal.toptal.net/clients
    - compare https://t-staff-portal-pr-12661.toptal.net/clients/841257 and https://staff-portal.toptal.net/clients/841257
  - Jest tests ✅ 
  - Cypress tests ✅ 
- Check out Client Portal pull request https://github.com/toptal/client-portal/pull/8965
  - Jest tests ✅ 
  - Cypress tests ✅ 
  - Visual tests (except for color of buttons) ✅ 
- Check out the branch and play around with grid, comparing it with https://picasso.toptal.net/
  - use https://github.com/sashuk/storybook-tabs-sync to simplify the comparison
 

https://github.com/toptal/picasso/assets/1390758/1574fc7e-4495-4fd7-9f59-e72b9ebbd868



### Development checks

- [N/A] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [N/A] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)
- [x] ⚠️ **Delete comments with** `DELETE_BEFORE_MERGE`
- [x] ⚠️ **Check if `@toptal/picasso-tailwind@2.4.0` is still the version from the future**

**Breaking change**

- [N/A] codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in Staff and Client Portals (please see the **How to test** section)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-4594]: https://toptal-core.atlassian.net/browse/FX-4594?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
